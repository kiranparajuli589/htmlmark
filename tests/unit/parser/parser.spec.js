import { commonTokensList, LinesToEscape } from "../../fixtures/commTokens.js"
import { MDP } from "../../../lib/index.js"


describe("Parser Commons", () => {
	describe("newline", () => {
		it.each([
			{ lines: [""] },
			{ lines: ["\n"] },
			{ lines: ["", "\n"] }
		])("should not include the top empty lines", ({ lines }) => {
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
		it.each(["", "\n"])("should include the empty line after some content", (eLine) => {
			const lines = [
				"some plain text",
				eLine
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
		it("should combine multiple consequent new lines to a single one", () => {
			const lines = [
				"some plain text",
				"\n",
				"",
				"some more plain text"
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
	})
	describe("comment", () => {
		it("should add a comment", () => {
			const lines = [
				"one",
				"<!-- comment -->",
				"three"
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
	})
	describe("common tokens", () => {
		it.each(commonTokensList)("should parse the common tokens", (line) => {
			const lines = [line]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
		it.each(commonTokensList)("should parse the common tokens with some space before", (line) => {
			const indent = "  "
			const lines = [indent + line]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
		it("should parse underlined heading", () => {
			const lines = [
				"Heading 1",
				"======",
				"Heading 2",
				"------",
				"Normal text"
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
	})
	describe("hr line", () => {
		it.each([
			"---",
			"--",
			"-----"
		])("should parse the hr line", (line) => {
			const lines = [
				line
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
		it.skip("should not allow multiple consecutive hr lines", () => {
			const lines = [
				"---",
				"---"
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
	})
	describe("escaping", () => {
		it("should escape everything inside a code block", () => {
			const lines = [
				"```",
				...LinesToEscape,
				"```"
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
		it("should escape everything inside a code", () => {
			const lines = [
				"`</em> & 'quo\"`"
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
		it("should escape everything excepts tags for other tokens", () => {
			const lines = [
				"# &Ampersand",
				"- 'SomeQuote'",
				"> Some \" comment",
				"<bold>bold</bold>"
			]
			const html = MDP.h(lines)
			expect(html).toMatchSnapshot()
		})
	})
})
