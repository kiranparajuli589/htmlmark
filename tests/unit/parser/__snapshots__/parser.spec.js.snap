// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser Commons comment should add a comment 1`] = `
"<p>one</p>
<!-- comment-->
<p>three</p>
"
`;

exports[`Parser Commons common tokens should parse the common tokens 1`] = `
"<h1>Heading One Text</h1>
"
`;

exports[`Parser Commons common tokens should parse the common tokens 2`] = `
"<h2>Heading Two Text</h2>
"
`;

exports[`Parser Commons common tokens should parse the common tokens 3`] = `
"<h3>Heading Three Text</h3>
"
`;

exports[`Parser Commons common tokens should parse the common tokens 4`] = `
"<h4>Heading Four Text</h4>
"
`;

exports[`Parser Commons common tokens should parse the common tokens 5`] = `
"<h5>Heading Five Text</h5>
"
`;

exports[`Parser Commons common tokens should parse the common tokens 6`] = `
"<h6>Heading Six Text</h6>
"
`;

exports[`Parser Commons common tokens should parse the common tokens 7`] = `
"<!-- comment item text-->
"
`;

exports[`Parser Commons common tokens should parse the common tokens 8`] = `"<img src='image.png' alt='alt text'>"`;

exports[`Parser Commons common tokens should parse the common tokens with some space before 1`] = `
"<h1>Heading One Text</h1>
"
`;

exports[`Parser Commons common tokens should parse the common tokens with some space before 2`] = `
"<h2>Heading Two Text</h2>
"
`;

exports[`Parser Commons common tokens should parse the common tokens with some space before 3`] = `
"<h3>Heading Three Text</h3>
"
`;

exports[`Parser Commons common tokens should parse the common tokens with some space before 4`] = `
"<h4>Heading Four Text</h4>
"
`;

exports[`Parser Commons common tokens should parse the common tokens with some space before 5`] = `
"<h5>Heading Five Text</h5>
"
`;

exports[`Parser Commons common tokens should parse the common tokens with some space before 6`] = `
"<h6>Heading Six Text</h6>
"
`;

exports[`Parser Commons common tokens should parse the common tokens with some space before 7`] = `
"<!-- comment item text-->
"
`;

exports[`Parser Commons common tokens should parse the common tokens with some space before 8`] = `"<img src='image.png' alt='alt text'>"`;

exports[`Parser Commons common tokens should parse underlined heading 1`] = `
"<h1>Heading 1</h1>
<h2>Heading 2</h2>
<p>Normal text</p>
"
`;

exports[`Parser Commons escaping should escape everything excepts tags for other tokens 1`] = `
"<h1>&amp;Ampersand</h1>
<ul>
  <li>&#39;SomeQuote&#39;</li>
</ul>
<blockquote>
  <p> Some &quot; comment</p>
<bold>bold</bold>

</blockquote>
"
`;

exports[`Parser Commons escaping should escape everything inside a code 1`] = `
"<p><code>&lt;/em&gt; &amp; &#39;quo&quot;</code></p>
"
`;

exports[`Parser Commons escaping should escape everything inside a code block 1`] = `
"<pre><code>
  &lt;div&gt;
&lt;/div&gt;
ab&lt;cd
efg&gt;
efg&lt;
&gt;&gt;&gt;
&lt;&lt;&lt;&lt;
&amp; ampersand
&#39;single quotes&#39;
&quot;double quotes&quot;
</code></pre>
"
`;

exports[`Parser Commons hr line should parse the hr line 1`] = `
"<hr>
"
`;

exports[`Parser Commons hr line should parse the hr line 2`] = `
"<p>--</p>
"
`;

exports[`Parser Commons hr line should parse the hr line 3`] = `
"<hr>
"
`;

exports[`Parser Commons newline should combine multiple consequent new lines to a single one 1`] = `
"<p>some plain text</p>
<p>some more plain text</p>
"
`;

exports[`Parser Commons newline should include the empty line after some content 1`] = `
"<p>some plain text</p>
"
`;

exports[`Parser Commons newline should include the empty line after some content 2`] = `
"<p>some plain text</p>
"
`;

exports[`Parser Commons newline should not include the top empty lines 1`] = `""`;

exports[`Parser Commons newline should not include the top empty lines 2`] = `""`;

exports[`Parser Commons newline should not include the top empty lines 3`] = `""`;
